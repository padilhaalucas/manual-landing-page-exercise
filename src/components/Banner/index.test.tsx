import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Banner from './index';

jest.mock('@/contexts/quiz.context', () => ({
  useQuiz: jest.fn(() => ({
    setShowQuiz: jest.fn(),
  })),
}));

describe('Components/Banner', () => {
  it('should render the logo', () => {
    render(<Banner />);
    const logo = screen.getByAltText('alto');
    expect(logo).toBeInTheDocument();
  });

  it('should render the main heading', () => {
    render(<Banner />);
    const heading = screen.getByText(/Be good to yourself/i);
    expect(heading).toBeInTheDocument();
  });

  it('should render the description', () => {
    render(<Banner />);
    const description = screen.getByText(/We're working around the clock/i);
    expect(description).toBeInTheDocument();
  });

  it('should render the "TAKE THE QUIZ" button', () => {
    render(<Banner />);
    const button = screen.getByText('TAKE THE QUIZ');
    expect(button).toBeInTheDocument();
  });

  it('should call setShowQuiz when the button is clicked', () => {
    const mockSetShowQuiz = jest.fn();
    jest.spyOn(require('@/contexts/quiz.context'), 'useQuiz').mockImplementation(() => ({
      setShowQuiz: mockSetShowQuiz,
    }));

    render(<Banner />);
    const button = screen.getByText('TAKE THE QUIZ');
    fireEvent.click(button);
    expect(mockSetShowQuiz).toHaveBeenCalledWith(true);
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<Banner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});