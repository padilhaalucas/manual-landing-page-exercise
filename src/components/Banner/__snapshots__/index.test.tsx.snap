// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Banner should match snapshot 1`] = `
<section
  className="bg-banner bg-cover bg-center bg-no-repeat min-h-screen w-full py-8 px-16 sm:px-20 md:px-24 lg:px-32 xl:px-36 flex flex-col"
>
  <div>
    <img
      alt="alto"
      className="w-8 h-8 sm:w-10 sm:h-10 md:w-12 md:h-12"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height={40}
      onError={[Function]}
      onLoad={[Function]}
      src="/assets/logo.svg"
      style={
        {
          "color": "transparent",
        }
      }
      width={40}
    />
  </div>
  <div
    className="mt-12 sm:mt-24 md:mt-30 lg:mt-32 w-full md:w-3/4 lg:w-2/3 xl:w-1/2"
  >
    <h1
      className="text-5xl sm:text-7xl xl:text-8xl font-semibold sm:font-medium mb-4 text-dark-green-manual leading-tight"
    >
      Be good 
      <br />
       to yourself
    </h1>
    <p
      className="mb-8 sm:mb-10 md:mb-12 lg:mb-8 text-xl sm:text-lg text-dark-green-manual w-10/12 sm:w-5/6 md:w-4/5"
    >
      We're working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.
    </p>
    <button
      className="bg-dark-red-manual text-white text-sm font-bold px-6 py-3 sm:px-8 sm:py-3 md:px-10 md:py-4 rounded hover:bg-opacity-90 transition-colors duration-300"
      onClick={[Function]}
    >
      TAKE THE QUIZ
    </button>
  </div>
</section>
`;
